xbmc/music/MusicDatabase.cpp

```diff
@@ -981,27 +999,16 @@ bool CMusicDatabase::GetAlbum(int idAlbum, CAlbum& album, bool getSongs /* = tru
     if (idAlbum == -1)
       return false; // not in the database
 
+    //Get album, song and album song info data using separate queries/datasets because we can have 
+    //multiple roles per artist for both album and songs and that makes a single combined join impractical
+    //Get album datae
     std::string sql;
-    if (getSongs)
-    {
-      sql = PrepareSQL("SELECT albumview.*,albumartistview.*,songview.*,songartistview.*,albuminfosong.* "
-                       "  FROM albumview "
-                       "  LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
-                       "  JOIN songview ON albumview.idAlbum = songview.idAlbum "
-                       "  LEFT JOIN songartistview ON songview.idSong = songartistview.idSong "
-                       "  LEFT JOIN albuminfosong ON albumview.idAlbum = albuminfosong.idAlbumInfo "
-                       "  WHERE albumview.idAlbum = %ld "
-                       "  ORDER BY albumartistview.iOrder, songview.iTrack, songartistview.iOrder", idAlbum);
-    }
-    else
-    {
-      sql = PrepareSQL("SELECT albumview.*,albumartistview.* "
-                       "  FROM albumview "
-                       "  LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
-                       "  WHERE albumview.idAlbum = %ld "
-                       "  ORDER BY albumartistview.iOrder", idAlbum);
-    }
-
+    sql = PrepareSQL("SELECT albumview.*,albumartistview.* "
+                     " FROM albumview "
+                     " LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
```

Comment:
Having checked the history JMarshall (who seems to have written much of this part) deliberately changed these to LEFT JOIN because albums and songs can exist without artists. See https://github.com/xbmc/xbmc/commit/70ee5f75a4d622757cb473fbb89c414ba1349592